sql/hooks/hooks.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2025 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hooks
+
+import (
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/plan"
+)
+
+// Global is a variable that contains the interface for calling hooks. By default, this contains no-op hooks as
+// integrators may implement their own hooks. This variable should be overwritten by the integrator.
+var Global Hooks
+
+// Hooks is an interface that represents various hooks that are called within a statement's lifecycle.
+type Hooks interface {
+	// Table returns hooks related to direct table statements.
+	Table() Table
```

Comment:
All of these should probably be plural, returning either a slice of hooks or an iterator over them. We should assume that there is going to be a chain of hooks that integrators set up to run in some order they choose.